ecommerce_extensions/payment/processors/edupay.py

```diff
@@ -0,0 +1,240 @@
+    @staticmethod
+    def get_single_seat(basket):
+        """
+        Return the first product encountered in the basket with the product
+        class of 'seat'.  Return None if no such products were found.
+
+        Arguments:
+            basket (Basket): Basket being purchased via the payment processor.
+
+        Returns:
+            product: First product with the class seat.
+        """
+        try:
+            seat_class = ProductClass.objects.get(slug='seat')
+        except ProductClass.DoesNotExist:
+            # this occurs in test configurations where the seat product class is not in use
+            return None
+
+        for line in basket.lines.all():
+            product = line.product
+            if product.get_product_class() == seat_class:
+                return product
+
+        return None
+
+    def handle_processor_response(self, response, basket=None):  # pylint: disable=unused-argument
+        """
+        Handle a response (i.e., "merchant notification").
+
+        This method does the following:
+            1. Verify the validity of the response.
+            2. Create PaymentEvents and Sources for successful payments.
+
+        Arguments:
+            response (dict): Dictionary of parameters received from the payment processor.
+
+        Keyword Arguments:
+            basket (Basket): Basket being purchased via the payment processor.
+
+        Raises:
+            TransactionDeclined: Indicates the payment was declined by the processor.
+            GatewayError: Indicates a general error on the part of the processor.
+            InvalidEdnxDecision: Indicates an unknown decision value
+
+        Returns:
+            HandleProcessorResponse: Payment info (transaction_id, total, currency, card_number and card_type)
+        """
+
+        # Validate the signature
+        if not self.is_signature_valid(response):
+            raise InvalidSignatureError
+        # Raise an exception for payments that were not accepted. Consuming code should be responsible for handling
+        # and logging the exception.
+        transaction_state = response['state_pol']
+        if transaction_state != self.TRANSACTION_ACCEPTED:
+            exception = {
+                self.TRANSACTION_DECLINED: TransactionDeclined,
+                self.TRANSACTION_ERROR: GatewayError,
+            }.get(transaction_state, InvalidEdnxDecision)
+
+            raise exception
+
+        currency = response.get('currency')
+        total = Decimal(response.get('value'))
+        transaction_id = response.get('transaction_id')
+        card_number = response.get('cc_number', '')
+        card_type = response.get('lapPaymentMethod', '')
+
+        return HandledProcessorResponse(
+            transaction_id=transaction_id,
+            total=total,
+            currency=currency,
+            card_number=card_number,
+            card_type=card_type,
+        )
+
+    def _generate_signature(self, parameters, signature_type):
```

Comment:
this block doesn't consider a default value so this will fail when signature_type is not PAYMENT_FORM_SIGNATURE or  CONFIRMATION_SIGNATURE